test/e2e/proxy_test.go

```diff
@@ -0,0 +1,283 @@
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"golang.org/x/net/html"
+	"io/ioutil"
+	mathrand "math/rand"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+func TestOAuthProxyE2E(t *testing.T) {
+	oauthProxyTests := map[string]struct {
+		appName            string
+		routeName          string
+		routePort          int32
+		servicePort        int32
+		serviceTargetPort  int
+		serviceAccountName string
+		certConfigMapName  string
+		oauthProxyArgs     []string
+		expectedErr        string
+	}{
+		"basic": {
+			appName:            "proxy",
+			routeName:          "proxy-route",
+			routePort:          int32(8443),
+			servicePort:        int32(443),
+			serviceTargetPort:  8443,
+			serviceAccountName: "proxy",
+			certConfigMapName:  "proxy-certs",
+			oauthProxyArgs: []string{
+				"--https-address=:8443",
+				"--provider=openshift",
+				"--openshift-service-account=proxy",
+				"--upstream=http://localhost:8080",
+				"--tls-cert=/etc/tls/private/tls.crt",
+				"--tls-key=/etc/tls/private/tls.key",
+				"--tls-client-ca=/etc/tls/private/ca.crt",
+				"--cookie-secret=SECRET",
+			},
+			expectedErr: "",
+		},
+		"scope-full": {
+			appName:            "proxy",
+			routeName:          "proxy-route",
+			routePort:          int32(8443),
+			servicePort:        int32(443),
+			serviceTargetPort:  8443,
+			serviceAccountName: "proxy",
+			certConfigMapName:  "proxy-certs",
+			oauthProxyArgs: []string{
+				"--https-address=:8443",
+				"--provider=openshift",
+				"--openshift-service-account=proxy",
+				"--upstream=http://localhost:8080",
+				"--tls-cert=/etc/tls/private/tls.crt",
+				"--tls-key=/etc/tls/private/tls.key",
+				"--tls-client-ca=/etc/tls/private/ca.crt",
+				"--cookie-secret=SECRET",
+				"--scope=user:full",
+			},
+			expectedErr: "403 Permission Denied",
+		},
+	}
+
+	image := os.Getenv("TEST_IMAGE")
```

Comment:
Failing if env vars isn't set is somewhat unusual in our testing.  I would probably expect a default value from the docker hub.